modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/healthcheck/declaration/PeerDeclarationHealthCheck.scala

```diff
@@ -116,10 +116,23 @@ object PeerDeclarationHealthCheck {
                   }
                 case (NegativeOutcome, round) =>
                   round.getRoundIds.flatMap { roundIds =>
-                    logger.info(s"Outcome for key ${key.show}: negative - removing facilitator | Round ids: ${roundIds.show}") >>
-                      consensusStorage.removeRegisteredPeer(key.id) >>
-                      (consensusStorage.addRemovedFacilitator(key.consensusKey, key.id) >>=
-                        consensusManager.checkForStateUpdateSync(key.consensusKey))
+                    consensusStorage
+                      .getRegisteredPeers(key.consensusKey)
+                      .map(_.contains(key.id))
+                      .ifM(
```

Comment:
This is not atomic, hence it's unsafe (check-then-act). Just between checking and removing a peer there could be a change to a registration list.

I propose a following solution, add another param to the `removeRegisteredPeer` so that it looks like this:
```
def removeRegisteredPeer(peerId: PeerId, key: Key): F[Unit] =
  peerRegistrationsR.update { peerRegistrations =>
    peerRegistrations
      .focus()
      .at(peerId)
      .modify(_.filter(_ <= key))
  }
```

It's also less code on the calling side.